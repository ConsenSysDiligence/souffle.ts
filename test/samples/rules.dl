// A,B,C,CanRenovate,Heritage,Housemate,LivesAt,Owner
// https://souffle-lang.github.io/rules#negation-in-rules
.decl Owner(person: symbol, building: symbol)
.decl Heritage(building: symbol)
.decl CanRenovate(person: symbol, building: symbol)

CanRenovate(person, building) :- Owner(person, building), !Heritage(building).

// https://souffle-lang.github.io/rules#multiple-heads
.decl A(x: number, y: number)
.decl B(x: number, y: number)
.decl C(x: number, y: number)

A(x,y), C(x,y) :- B(x,y). 

// https://souffle-lang.github.io/rules#disjunction
.decl Housemate(personA: symbol, personB: symbol)
.decl LivesAt(person: symbol, building: symbol)

LivesAt(person, building) :-
    Owner(owner, building),
    ( person=owner ; Housemate(owner, person) ).
