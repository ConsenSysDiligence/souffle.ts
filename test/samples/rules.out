/// A
===============
x y
/// B
===============
x y
/// C
===============
x y
/// CanRenovate
===============
person building
/// Heritage
===============
building
/// Housemate
===============
personA personB
/// LivesAt
===============
person building
/// Owner
===============
person building